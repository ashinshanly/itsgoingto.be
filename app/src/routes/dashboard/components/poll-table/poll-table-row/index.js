import PollTableRow from './poll-table-row'

export default PollTableRow
