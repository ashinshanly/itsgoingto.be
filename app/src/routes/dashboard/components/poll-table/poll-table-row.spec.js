/* global expect */
import React from 'react'
import { shallow } from 'enzyme'
import { browserHistory } from 'react-router'
import { PollTableRow } from './poll-table-row'

browserHistory.push = jest.fn()

const props = {
  poll: {
    id: '1',
    identifier: 'sk93jn28d',
    question: 'This is a question',
    responsesCount: 5,
    deleted: false,
    ended: false,
    created: {
      date: '2019-02-23 13:45:37.000000',
    },
  },
  deletePoll: jest.fn(),
}
let wrapper

describe('(Route) dashboard', () => {
  describe('(Component) poll table row', () => {
    beforeEach(() => {
      wrapper = shallow(<PollTableRow {...props} />)
    })

    afterEach(() => {
      jest.clearAllMocks()
    })

    describe('(Action) identifier onClick', () => {
      it('should route to identifier', () => {
        wrapper.find('#identifier-1').simulate('click')
        expect(browserHistory.push).toHaveBeenCalledTimes(1)
        expect(browserHistory.push).toHaveBeenCalledWith('/sk93jn28d')
      })
    })

    describe('(Action) delete onClick', () => {
      it('should set state deleting true', () => {
        wrapper.find('#delete-1').simulate('click')
        expect(wrapper.state().deleting).toBe(true)
      })

      it('should call deletePoll with poll identifier', () => {
        wrapper.find('#delete-1').simulate('click')
        expect(props.deletePoll).toHaveBeenCalledTimes(1)
        expect(props.deletePoll).toHaveBeenCalledWith('sk93jn28d')
      })
    })

    describe('(Render)', () => {
      describe('poll is active', () => {
        it('matches snapshot', () => {
          wrapper = shallow(<PollTableRow {...props} />)
          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('poll is ended', () => {
        it('matches snapshot', () => {
          wrapper = shallow(<PollTableRow {...props} poll={{ ...props.poll, ended: true }} />)
          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('poll is deleted', () => {
        it('matches snapshot', () => {
          wrapper = shallow(<PollTableRow {...props} poll={{ ...props.poll, deleted: true }} />)
          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('poll is deleting', () => {
        it('matches snapshot', () => {
          wrapper.setState({ deleting: true })
          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })
})