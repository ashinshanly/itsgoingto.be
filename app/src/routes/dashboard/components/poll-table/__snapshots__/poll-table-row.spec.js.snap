// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) dashboard (Component) poll table row (Render) poll is active matches snapshot 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    <a
      id="identifier-1"
      onClick={[Function]}
    >
      sk93jn28d
    </a>
  </td>
  <td>
    This is a question
  </td>
  <td>
    5
  </td>
  <td>
    Active
  </td>
  <td>
    23-02-2019 13:45
  </td>
  <td>
    <span>
      <a
        id="delete-1"
        onClick={[Function]}
      >
        <i
          className="fa fa-times"
        />
      </a>
    </span>
  </td>
</tr>
`;

exports[`(Route) dashboard (Component) poll table row (Render) poll is deleted matches snapshot 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    <a
      id="identifier-1"
      onClick={[Function]}
    >
      sk93jn28d
    </a>
  </td>
  <td>
    This is a question
  </td>
  <td>
    5
  </td>
  <td>
    Deleted
  </td>
  <td>
    23-02-2019 13:45
  </td>
  <td />
</tr>
`;

exports[`(Route) dashboard (Component) poll table row (Render) poll is deleting matches snapshot 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    <a
      id="identifier-1"
      onClick={[Function]}
    >
      sk93jn28d
    </a>
  </td>
  <td>
    This is a question
  </td>
  <td>
    5
  </td>
  <td>
    Active
  </td>
  <td>
    23-02-2019 13:45
  </td>
  <td>
    <span>
      <i
        className="fa fa-circle-o-notch fa-spin"
      />
    </span>
  </td>
</tr>
`;

exports[`(Route) dashboard (Component) poll table row (Render) poll is ended matches snapshot 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    <a
      id="identifier-1"
      onClick={[Function]}
    >
      sk93jn28d
    </a>
  </td>
  <td>
    This is a question
  </td>
  <td>
    5
  </td>
  <td>
    Ended
  </td>
  <td>
    23-02-2019 13:45
  </td>
  <td>
    <span>
      <a
        id="delete-1"
        onClick={[Function]}
      >
        <i
          className="fa fa-times"
        />
      </a>
    </span>
  </td>
</tr>
`;
